utils.go

```diff
@@ -9,3 +9,11 @@ func allHealthy(results map[string]Result) (healthy bool) {
 
 	return true
 }
+
+func copyResultMap(result map[string]Result) map[string]Result {
```

Comment:
There's an health.Results() func. Why do we need this util too?

Reply:
I'm copying the map so the original one won't get changed by listeners